nanoFirmwareFlasher.Library/OutputWriter.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Threading;
+
+namespace nanoFramework.Tools.FirmwareFlasher
+{
+    /// <summary>
+    /// Writes output to the console. In unit tests the output can be directed to
+    /// a per-test implementation to make the library and tool testable.
+    /// This is the easiest way, as library and tool were infested with Console.*
+    /// that is hard to test, especially when running tests in parallel.
+    /// </summary>
+    public static class OutputWriter
+    {
+        #region Fields
```

Comment:
Why are we adding these regions? This is not a common use on nano code base... In particular here, there are only one field and a property...

Reply:
I use a lot of regions, it helps me navigating a code file. I usually have most  regions collapsed and only those expanded that I'm actually working on. I also use regions inside methods so that the method. If you don't like it that way, tell me how you want it.

Reply:
Understood. That's a personal preference and that's OK.
Can you please add an empty line before/after the regions?

Its soooo much easier to read the code when there are empty lines separating what's not relevant to the code (in this case the regions start/end declarations).
Same goes with the rest of the code. I find it much easier going through the code with the eyes if the code is split into "chuncks" to hint the though process about those.